lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -397,13 +397,35 @@ def _is_supported_index(df_index: pd.Index) -> bool:
     )
 
 
+def _fix_column_headers(data_df: pd.DataFrame) -> None:
+    """Fix the column headers of the provided dataframe inplace to work
+    correctly for data editing."""
+
+    if isinstance(data_df.columns, pd.MultiIndex):
+        # Flatten hierarchical column headers to a single level:
+        data_df.columns = [
+            "_".join(str(header)) for header in data_df.columns.to_flat_index()
```

Comment:
Shouldn't `str` be applied to each element of `header` rather than to `header` itself (which should be a tuple)?

Reply:
Yep indeed. That was a stupid last-minute change 🫣 I added a snapshot test to cover this.